pkg/router/controller/status_test.go

```diff
@@ -373,11 +395,253 @@ func TestStatusRecordRejectionConflict(t *testing.T) {
+							Status:             kapi.ConditionFalse,
+							LastTransitionTime: &touched1,
+						},
+					},
+				},
+			},
+		},
+	})
+
+	checkResult(t, err, c1, admitter1, "route2.test-new.local", now2, &now2.Time, 1, 1)
+}
+
+func makePass(t *testing.T, host string, admitter *StatusAdmitter, srcObj *routeapi.Route, expectUpdate bool, conflict bool) *routeapi.Route {
+	// initialize a new client
+	var c *testclient.Fake
+	if conflict {
+		c = testclient.NewSimpleFake(&(errors.NewConflict(kapi.Resource("Route"), "route1", nil).(*errors.StatusError).ErrStatus))
+	} else {
+		c = testclient.NewSimpleFake(&routeapi.Route{})
+	}
+
+	admitter.client = c
+
+	inputObjRaw, err := kapi.Scheme.DeepCopy(srcObj)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+
+	inputObj := inputObjRaw.(*routeapi.Route)
+	inputObj.Spec.Host = host
+
+	err = admitter.HandleRoute(watch.Modified, inputObj)
+
+	if expectUpdate {
+		now := nowFn()
+		var nowTime *time.Time
+		if !conflict {
+			nowTime = &now.Time
+		}
+		return checkResult(t, err, c, admitter, inputObj.Spec.Host, now, nowTime, 0, 0)
+	} else {
+		if err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		// expect the last HandleRoute not to have performed any actions
+		if len(c.Actions()) != 0 {
+			t.Fatalf("unexpected actions: %#v", c)
+		}
+
+		return nil
+	}
+}
+
+// This test tries an extended interaction between two "old" and two "new"
+// router replicas.
+func TestProtractedStatusFightBetweenRouters(t *testing.T) {
+	p := &fakePlugin{}
+
+	oldHost := "route1.test.local"
+	newHost := "route1.test-new.local"
+	oldHost2 := "route2.test.local"
+	newHost2 := "route2.test-new.local"
+	newHost3 := "route3.test-new.local"
+
+	now := unversioned.Now()
+	nowFn = func() unversioned.Time { return now }
+
+	initObj := &routeapi.Route{
+		ObjectMeta: kapi.ObjectMeta{Name: "route1", Namespace: "default", UID: types.UID("uid1")},
+		Spec:       routeapi.RouteSpec{Host: newHost},
+		Status:     routeapi.RouteStatus{},
+	}
+
+	// NB: contention period is 1 minute
+
+	newAdmitter1 := NewStatusAdmitter(p, nil, "test")
+	newAdmitter2 := NewStatusAdmitter(p, nil, "test")
+
+	oldAdmitter1 := NewStatusAdmitter(p, nil, "test")
+	oldAdmitter2 := NewStatusAdmitter(p, nil, "test")
+
+	t.Logf("Setup up the two 'old' routers")
+	currObj := makePass(t, oldHost, oldAdmitter1, initObj, true, false)
+	makePass(t, oldHost, oldAdmitter2, currObj, false, false)
+
+	t.Logf("Phase in the two 'new' routers (with the second getting a conflict)...")
+	now = unversioned.Time{Time: now.Add(5 * time.Second)}
+	nowFn = func() unversioned.Time { return now }
+
+	makePass(t, newHost, newAdmitter2, currObj, true, true)
+	currObj = makePass(t, newHost, newAdmitter1, currObj, true, false)
+
+	t.Logf("...which should cause 'new' router #2 and the two 'old' routers to receive an update, and ignore it")
+	now = unversioned.Time{Time: now.Add(1 * time.Second)}
+	nowFn = func() unversioned.Time { return now }
```

Comment:
nit: I wouldn't expect you to have to redefine this since the closure should capture the variable.